pkg/refs/refs.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2019 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package refs
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func NewTypedLocalObjectReference(name string, gk schema.GroupKind) *TypedLocalObjectReference {
+	if name == "" || gk.Empty() {
+		return nil
```

Comment:
Is it really sound to return nil in all those edge cases rather than say, panic? Seems like we never check for nil at call sites

Reply:
I agree this is being overly defensive.